stdlib/messaging/activemq-artemis/src/main/java/org/ballerinalang/messaging/artemis/ArtemisTransactionContext.java

```diff
@@ -21,34 +21,48 @@
 
 import org.apache.activemq.artemis.api.core.ActiveMQException;
 import org.apache.activemq.artemis.api.core.client.ClientSession;
+import org.ballerinalang.jvm.Strand;
+import org.ballerinalang.jvm.transactions.BallerinaTransactionContext;
+import org.ballerinalang.jvm.transactions.TransactionLocalContext;
+import org.ballerinalang.jvm.transactions.TransactionResourceManager;
+import org.ballerinalang.jvm.util.exceptions.BallerinaException;
 import org.ballerinalang.jvm.values.ObjectValue;
-import org.ballerinalang.util.transactions.BallerinaTransactionContext;
 
-import javax.transaction.xa.XAResource;
+import java.util.Objects;
+import java.util.UUID;
 
-//Todo: fix this class when transaction supported
+import javax.transaction.xa.XAResource;
 
 /**
  * Session wrapper class with Ballerina transaction logic.
  */
 public class ArtemisTransactionContext implements BallerinaTransactionContext {
-//    private final String connectorId;
+    private final String connectorId;
     private ClientSession session;
     private ObjectValue sessionObj;
 
     public ArtemisTransactionContext(ObjectValue sessionObj) {
         this.sessionObj = sessionObj;
         session = (ClientSession) sessionObj.getNativeData(ArtemisConstants.ARTEMIS_SESSION);
-//        connectorId = UUID.randomUUID().toString();
+        connectorId = UUID.randomUUID().toString();
     }
 
     @Override
     public void commit() {
-
+        try {
+            session.commit();
+        } catch (ActiveMQException e) {
+            throw new BallerinaException("Transaction commit failed: " + e.getMessage(), e);
```

Comment:
I think we need to throw `ErroValue` for trapping

Reply:
Fixed in 687ca736ec93176ebd0358041c3a6913debf847f